### PR TITLE
Generate Screenshots in CI

### PR DESCRIPTION
**Pull request by lcreid**
Generate screenshots in CI, and add a commit if any screenshots change. Add some documentation that explains how to deal with screenshots changing.

There are a couple of examples (50, 51) that appear to be failing. I've added #714 for them. 

I put the sequence of steps to commit the changes in a Rake task, thinking it would allow us to run the identical steps locally, if needed. Also, it's a bit easier to handle the long commit message in Ruby, at least for me. Does that make sense?

The `git push` when screenshots change triggers another CI run. To avoid that, I added `[skip ci]` to the commit message, but I fear that it would prevent CI runs when, for example, one rebases on the upstream main. The last commit will still have `[skip ci]` and so won't run. So I took it out and we'll simply live with double CI. Okay?

Fixes: #711

### CODE REVIEW

**Comment by lcreid** [comment]
@donv This might affect your workflow more than mine. What do you think?

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -56,3 +56,8 @@ jobs:
       - name: Run tests
         working-directory: demo
         run: bundle exec rake test:all
+      - name: Commit changed screenshots and other files
+        if: failure()
+        working-directory: demo
+        run: bundle exec rake commit
+
```
Comment (donv) [inline .github/workflows/ruby.yml:63]: This will automatically commit all changed files in the demo directory after a failure?  That does not seem right.

Comment (lcreid) [reply .github/workflows/ruby.yml:63]: I couldn't think of a better way to communicate the changed screenshots back into a place where the dev could review the screenshots and see if their code needed to be fixed. GitHub's diff for the screenshots is actually pretty slick. I admit that the revert step may be a nuisance, or people may forget it. But we should still catch changes when reviewing the PR.

I could make the workflow more specific, so we _only_ save the screenshot changes when it was the screenshot diff that failed.

I'm open for other suggestions about how to generate and capture the screenshots when we create new ones, or the change is intended.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -50,6 +50,26 @@ You may find the [demo application](#the-demo-application) useful for developmen
...
 
+When you create or update a pull request, GitHub automatically runs tests that generate the screenshots in the [`README.md`](/README.md). If any of the screenshots change, GitHub will add an additional commit with the updated screenshots.
+
+Normally, the screenshots should _not_ change. If the screenshots changed, please review them _carefully_. If something changed that shouldn't have:
+
+```bash
+git pull # to bring the additional commit to your local branch.
+git revert HEAD # to remove the changes.
+```
+
```
[the remark this answers is not in the document]
Comment (lcreid) [reply CONTRIBUTING.md:61]: Really good point! Thanks!

**Comment by lcreid** [comment]
I'll merge this and we can discuss better solutions later.

**Pull request merged**